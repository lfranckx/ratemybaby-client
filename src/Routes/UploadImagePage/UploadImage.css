#upload-section {
    display: inline-block;
    background-color: white;
    border-radius: 10px;
    padding: 40px;
    box-shadow: 1.5px 3px 5px grey;
    width: 400px;
    color: rgb(41, 39, 39);
}

#upload-header {
    font-size: 2.6rem;
    opacity: 0.9;    
}

#upload-form > * {
    margin: 10px;
}

#upload-section label {
    color: rgb(41, 39, 39);
}

#upload-button {
    margin-top: 30px;
    background-color: rgb(249, 103, 103);
    background-image: linear-gradient(to right, rgb(247, 87, 87) , rgb(249, 147, 103));    
    display: inline-block;
    margin-top: 10px;
    padding: 10px 30px;
    border-radius: 50px;
    border: none;
    color: white;
    cursor: pointer;
}

#upload-button:active {
    background-color: rgb(249, 103, 103);
    background-image: linear-gradient(to right, rgb(247, 87, 87) , rgb(247, 87, 87)); 
}

/* MEDIA QUERIES */

@media (max-width: 700px) {
    #upload-section {
        width: 320px;
    }
    
    #upload-header {
        font-size: 1.5rem;
    }

    #upload-section label {
        font-size: 1rem;
    }

    #upload-section input {
        font-size: 1rem;
    }

    #upload-button {
        font-size: 15px;
        padding: 6.5px 35px;
    }
}