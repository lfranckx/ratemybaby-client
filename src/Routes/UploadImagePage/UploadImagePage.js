import React, { Component } from 'react'
import UploadImageForm from '../../Components/UploadImageForm/UploadImageForm'
import BabyContext from '../../Contexts/BabyContext'
import BabyApiService from '../../Services/baby-api-service'
import './UploadImage.css'

export default class UploadImagePage extends Component {
    static defaultProps = {
        location: {},
        history: {
          push: () => {},
        },
    }

    static contextType = BabyContext

    componentWillUnmount() {
        BabyApiService.getByParentId()
        .then(res => {
            this.context.setUsersBabies(res)
        })
        .catch(this.context.setError)
    }

    handleUploadSuccess = () => {
        const { baby } = this.context
        const { location, history } = this.props
        const destination = (location.state || {}).from || `/profile/${baby.id}`
        history.push(destination)
    }

    render() {
        console.log('UploadImage props', this.props);
        console.log('UploadImage context', this.context);
        
        const { baby } = this.context
        return (
            <section id="upload-section">
                <h3 id="upload-header">Upload Image</h3>
                    <UploadImageForm 
                        onUploadSuccess={this.handleUploadSuccess}
                        baby={baby}    
                    />
            </section>
        )
    }
}