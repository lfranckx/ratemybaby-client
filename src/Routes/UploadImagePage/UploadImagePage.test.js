import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import UploadImagePage from './UploadImagePage';

const context = {
    active: false,
    usersBabies: [],
    babies: [],
    baby: {},
    error: null,
    setError: () => {},
    clearError: () => {},
    toggleActive: () => {},
    setNotActive: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    setUsersBabies: () => {}
}

describe(`UploadImagePage component`, () => {

    it(`renders upload page without crashing`, () => {
        const wrapper = mount(<UploadImagePage />, { context: context })
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})