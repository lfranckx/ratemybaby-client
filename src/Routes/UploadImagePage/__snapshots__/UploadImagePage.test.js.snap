// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImagePage component renders upload page without crashing 1`] = `
<UploadImagePage
  history={
    Object {
      "push": [Function],
    }
  }
  location={Object {}}
  match={
    Object {
      "params": Object {},
    }
  }
>
  <section
    id="upload-section"
  >
    <h2
      id="upload-header"
    >
      Upload Image
    </h2>
    <UploadImage
      baby={
        Object {
          "baby": null,
        }
      }
      onLoginSuccess={[Function]}
      onUploadSuccess={[Function]}
    >
      <form
        id="upload-form"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <label>
          Select an image to upload:
        </label>
        <input
          accept=".png, .jpg, .jpeg .gif"
          aria-label="file"
          id="file"
          name="file"
          type="file"
        />
        <button
          id="upload-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </UploadImage>
  </section>
</UploadImagePage>
`;
